### ISSUE TITLE
import: express priority in syntax

### ISSUE THREAD

**Issue by shermanCRL** [cockroachdb/cockroach#81818]
#### Background

IMPORT is designed to have priority lower than foreground SQL traffic, via admission control, in v22.1. (This priority is called [`BulkNormalPri`](https://github.com/cockroachdb/cockroach/blob/release-22.1/pkg/util/admission/work_queue.go/#L148-L167), which is not a publicly accessible thing.)

Users often ask about optimal [cluster settings](https://www.cockroachlabs.com/docs/stable/cluster-settings.html) when they want to improve the [performance of imports](https://www.cockroachlabs.com/docs/stable/import-performance-best-practices.html). The behavior of those settings can be complex / hard to predict. It’s also easy to forget to set them back to their original values after the import.

#### Suggestion

Allow users to express ‘maximize IMPORT performance’ via syntax, instead of having to figure out cluster settings. Something like `IMPORT ... WITH priority=‘high’` or similar.

This option might upgrade the import job to run at a higher [priority](https://github.com/cockroachdb/cockroach/blob/release-22.1/pkg/util/admission/work_queue.go#L143-L167) in admission control. It might also temporarily change cluster settings such as `kv.bulk_io_write.concurrent_addsstable_requests`, and return them to their previous value after the job.

To be clear, we should design our default behavior to be optimal for most users -- IMPORT should be as fast as possible while minimizing impact on foreground traffic. The above syntax expresses an operator’s willingness to degrade other traffic in exchange for faster import.

Jira issue: CRDB-16083

**Comment by shermanCRL** [comment]
After discussing with @dt, this configurability is not likely to be an effective or desirable feature in real scenarios. Better to focus on maturing Admission Control in general, as well as overall IMPORT performance.

**Comment by dt** [comment]
I’m not as optimistic on trying to doc how to tune IMPORT or other ingest performance since it is a question with no “right” answer and mostly just ends up as “try it and see”. I’m more optimistic about ripping out a bunch of the existing tuning knobs and letting the system pace itself for maximum safe utilization, based on its own measurements of the actual system, than I am on trying to tell operators how to tune it.

basically: https://github.com/cockroachdb/cockroach/issues/81772

**Comment by mwang1026** [comment]
Ya. I guess, specifically, "don't run foreground traffic and/or watch it closely" -- didn't mean we should talk about tweaking knobs.
